### PR TITLE
[ UI component ]: Remove jQuery from all components

### PR DESCRIPTION
**Pull request by rogeruiz**
## Description

Remove jQuery as per #1351 

## Additional information

* #1351 

Before you hit Submit, make sure you’ve done whichever of these applies to you:

- [x] Follow the [18F Front End Coding Style Guide](https://pages.18f.gov/frontend/) and [Accessibility Guide](https://pages.18f.gov/accessibility/checklist/).
- [x] Run `npm test` and make sure the tests for the files you have changed have passed.
- [ ] Run your code through [HTML_CodeSniffer](http://squizlabs.github.io/HTML_CodeSniffer/) and make sure it’s error free.
- [x] Title your pull request using this format: [Website] - [UI component]: Brief statement describing what this pull request solves.

## Acceptance Criteria

- [x] Evaluate how jQuery is used for the UI components.
- [x] Rewrite any jQuery specific code to be compatible with current browser support for the standards.
    - [x] `utils/select.js` for selecting elements from the DOM.
    - [x] `utils/dispatch.js` for attaching events to DOM elements and the window.
    - [x] `utils/is-dom-ready.js` for checking if the `DOMLoadedContent` event has fired.
    - [x] `utils/verify-jquery.js` for checking if `jQuery` is loaded for any legacy code that **requires** it.
- [x] Test within all currently supported browsers to ensure no regressions are introduced.
- [x] Document / Create any issues that were discovered during this task.

### CODE REVIEW

File: src/js/components/accordion.js [hunk 1]
```diff
@@ -9,24 +9,30 @@ var $ = require('jquery');
  * @param {jQuery} el A jQuery html element to turn into an accordion.
  */
 function Accordion ($el) {
-  var self = this;
-  this.$root = $el;
+  var self = this; // Node
+  this.$root = $el; // jQuery powered element
 
   // delegate click events on each <button>
+  //         v------- utils/dispatch
```
Comment (msecret) [inline src/js/components/accordion.js]: What are these comments? Are they a special syntax for documentation or something?

Comment (rogeruiz) [inline src/js/components/accordion.js]: Nah. They're just notes for what utilities will be replacing jQuery methods

**Comment by rogeruiz** [comment]
@juliaelman Please take a look at this and help us tick the last checkbox in the first comment.

File: docs/_layouts/styleguide.html [hunk 2]
```diff
@@ -70,7 +70,7 @@ <h1 id="{{ page.title | slugify }}">{{ page.title }}</h1>
     {% include ie-polyfill-scripts.html %}
     {% include analytics.html %}
     {% include docs-scripts.html %}
-    <script src="{{ site.baseurl }}/assets/js/vendor/uswds.min.js"></script>
+    <script src="{{ site.baseurl }}/assets/js/vendor/uswds.js"></script>
```
Comment (juliaelman) [inline docs/_layouts/styleguide.html]: Was this changed for debugging purposes?

**Comment by juliaelman** [comment]
Nice work! Here is where I saw some errors lurking:

###  All browsers - /footers/

```
uswds.js
ReferenceError: _ is not defined
```

### Internet Explorer - ALL
```
Unmatched end tag
accordions 465, 385
accordions 496, 5
```

### Internet Explorer 10, 9 
```
accordions.js 11,1
SCRIPT438: OBJECT DOESN'T SUPPORT PROPERTY OR METHOD 'PREVENTDEFAULT'
```

### Internet Explorer 9
```
SCRIPT5009: 'requestAnimationFrame is undefined
styleguide.js (3, 31439)'
```

**Comment by rogeruiz** [comment]
Hmm that `styleguide.js` Seems out of scope for these changes. I don't think the jQuery removal touched anything in there.

**Comment by juliaelman** [comment]
@rogeruiz agreed that it is out of scope for this PR and have removed from the list.

File: docs/_layouts/styleguide.html (hunk 2 above)
Comment (rogeruiz) [inline docs/_layouts/styleguide.html]: Yes just for debuggability. We should revert this @yozlet

File: config/gulp/website.js [hunk 3]
```diff
@@ -54,7 +54,7 @@ gulp.task('copy-bundled-javascript', function (done) {
 
   dutil.logMessage('copy-bundled-javascript', 'Copying ' + dutil.dirName + ' release JavaScript');
 
-  return gulp.src('dist/js/' + dutil.pkg.name + '.min.js')
+  return gulp.src('dist/js/' + dutil.pkg.name + '.js')
```
Comment (rogeruiz) [inline config/gulp/website.js]: Same goes for this line. It needs to be reverted.

**Comment by yozlet** [comment]
@juliaelman I think I've fixed everything - can you give it another look? Thank you!

**Comment by juliaelman** [comment]
@yozlet perfect - nice work folks! 🚀

**Pull request merged**